File: backend/Strategy/Order.rb
require_relative '../ValueObjects/Address.rb'
require_relative '../ValueObjects/OrderItem.rb'

class Order
    attr_reader :customer, :product, :items, :payment, :address, :closed_at
  
    def initialize(customer, overrides = {})
      @customer = customer
      @items = []
      @order_item_class = overrides.fetch(:item_class) { OrderItem }
    end
  
    def apply_payment(pay_method)
      @payment = Payment.new(order: self, payment_method: CreditCard.fetch_by_hashed('43567890-987654367'))
      @payment.pay
    end

    def add_product(product)
      @items << @order_item_class.new(order: self, product: product)
    end
  
    def total_amount
      @items.map(&:total).inject(:+)
    end
  
    def close(closed_at = Time.now)
      @closed_at = closed_at
    end

    def closed?
      !closed_at.nil?
    end

    def post_payment
        raise "Not Implemented Exception"
    end
    # remember: you can create new methods inside those classes to help you create a better design
  end

File: backend/Solucao Creditas Challenge/PythonApplication1/order.py
import time
from customer import Address
from customer import Customer
from product import Product

class OrderTotal:
    closed_at = None
    items = []
    totalPrice = 0.0
    numOfItems = None
    
    def __init__(self, Customer):
        self.customer = Customer
        self.items = []
        self.numOfItems = None
        self.totalPrice = 0.0

    def putOnCart(self, OrdemItem):
        self.items.append([OrdemItem.order, OrdemItem.productType])
        self.numOfItems = len(self.items)
        self.totalPrice += OrdemItem.price
   
    def close(self, closed_at=time.time()):
        self.closed_at = closed_at
        Product(self.items, self.customer)

    # remember: you can create new methods inside those classes to help you create a better design

class OrderItem():
    order = None
    price = 0.0
  
    def __init__(self, order, productType, price):
        self.order = order
        self.productType = productType
        self.price = price